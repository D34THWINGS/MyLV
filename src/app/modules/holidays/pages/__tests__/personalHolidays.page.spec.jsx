import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import { PersonalHolidaysPage } from '../personalHolidays.page'

jest.mock('moment', () => jest.fn(date => ({ format: jest.fn(() => date) })))
jest.unmock('../personalHolidays.page')

describe('PersonalHolidaysPage', () => {
  let props
  beforeEach(() => {
    props = {
      fetchPersonalHolidays: jest.fn(),
      push: jest.fn(),
      deleteHoliday: jest.fn(),
      isLoading: false,
      classes: {},
      holidays: [{
        id: 'foo',
        date: 'bar',
        status: 'pending',
        periods: [{ baz: 'qux' }],
      }, {
        id: 'baz',
        date: 'qux',
        status: 'validated',
        periods: [],
      }],
    }
  })

  const getWrapper = () => shallow(<PersonalHolidaysPage {...props} />)

  it('should render properly', () => {
    // When
    const wrapper = getWrapper()

    // Then
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('should render in loading state', () => {
    // Given
    props.isLoading = true

    // When
    const wrapper = getWrapper()

    // Then
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
