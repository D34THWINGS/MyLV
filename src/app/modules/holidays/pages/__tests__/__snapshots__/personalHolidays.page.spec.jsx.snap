// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalHolidaysPage should render in loading state 1`] = `<WithStyles(LoadingPage) />`;

exports[`PersonalHolidaysPage should render properly 1`] = `
<WithStyles(Paper)>
  <WithStyles(Toolbar)>
    <WithStyles(Typography)
      component="h2"
      gutterBottom={true}
      variant="headline"
    >
      Mes demandes de congés
    </WithStyles(Typography)>
  </WithStyles(Toolbar)>
  <div>
    <WithStyles(Table)>
      <TableHead
        component="thead"
      >
        <WithStyles(TableRow)>
          <WithStyles(TableCell)>
            Titre
          </WithStyles(TableCell)>
          <WithStyles(TableCell)>
            Date de la demande
          </WithStyles(TableCell)>
          <Hidden
            implementation="js"
            lgDown={false}
            lgUp={false}
            mdDown={true}
            mdUp={false}
            smDown={false}
            smUp={false}
            xlDown={false}
            xlUp={false}
            xsDown={false}
            xsUp={false}
          >
            <WithStyles(TableCell)
              key="paidHolidays"
              numeric={true}
            >
              Congés payés
            </WithStyles(TableCell)>
            <WithStyles(TableCell)
              key="unpaidHolidays"
              numeric={true}
            >
              Congés sans solde
            </WithStyles(TableCell)>
            <WithStyles(TableCell)
              key="conventionalHolidays"
              numeric={true}
            >
              Absences Syntec
            </WithStyles(TableCell)>
            <WithStyles(TableCell)
              numeric={true}
            >
              Status
            </WithStyles(TableCell)>
          </Hidden>
        </WithStyles(TableRow)>
      </TableHead>
      <TableBody
        component="tbody"
      >
        <WithStyles(HolidayRow)
          holiday={
            Object {
              "date": "bar",
              "id": "foo",
              "periods": Array [
                Object {
                  "baz": "qux",
                },
              ],
              "status": "pending",
            }
          }
          key="foo"
          onClick={[Function]}
          onHolidayDelete={[MockFunction]}
        />
        <WithStyles(HolidayRow)
          holiday={
            Object {
              "date": "qux",
              "id": "baz",
              "periods": Array [],
              "status": "validated",
            }
          }
          key="baz"
          onClick={[Function]}
          onHolidayDelete={[MockFunction]}
        />
      </TableBody>
    </WithStyles(Table)>
  </div>
  <WithStyles(CardActions)>
    <WithStyles(Button)
      color="primary"
      component={[Function]}
      size="small"
      to="/holidays/new"
    >
      Nouvelle demande
    </WithStyles(Button)>
  </WithStyles(CardActions)>
</WithStyles(Paper)>
`;
